### ISSUE TITLE
Colliders on the new LC tanks (bare)

### ISSUE THREAD

**Issue by taterkerman** [shadowmage45/SSTULabs#604]
![lc collider 1](https://user-images.githubusercontent.com/17967275/32740212-6de21d68-c85f-11e7-89a3-06e4b5fae047.png)
As placed from the parts menu, the tanks are set such that with 4 symmetry the legs are between tanks, floating.

Rotated such that the legs should land on the tanks, the colliders don't:

![lc collider 2](https://user-images.githubusercontent.com/17967275/32740271-a23038de-c85f-11e7-9271-3098372e3ce5.png)

Hard to tell in my bad pic, but I shift "e" keyed the tanks until they were exactly aligned, and the legs are on the tank, but are NOT centered on each tank, even though the tanks in question are on the corners.

Not sure what can be done to make legs that play nice with this awesome tank. Obviously, you can clip them in using that tool in the VAB, but the tool doesn't play nice with symmetry, so you have to then place each leg individually (yuck! I always have them slightly off when I try this).

**Comment by shadowmage45** [comment]
You are aware that the offset-tool has a 'global' and 'local' mode?  (press the 'F' key to switch).  In local mode, you can offset the legs with perfect accuracy.  (global/local also applies to the rotation tool)

I.E.  This isn't a problem?  I've had no problem with positioning landing legs even when they were placed using 4x radial symmetry.

**Comment by shadowmage45** [comment]
As to the root of the problem, the colliders not matching the geometry very precisely -- yep, fully intentional.  It would be 'bad' for performance to have form-fitting colliders for such complex shapes.  As such the tanks all use an octagonal collider (or 8x cuboids for the 'hollow' variants).

If someone wanted to put together some performance tests comparing various collider setups, and prove to me that it wouldn't drag performance down too bad, I would be willing to use a more accurate collider setup.  I don't even have time to do the testing though, and certainly don't have time to redo all of the colliders without knowing unquestionably that it will all work out.

**Comment by taterkerman** [comment]
I just tested this. I placed an Orion. I put the new tank below. When freshly placed, the gap between 2 of the tanks is exactly centered on the windows (the pillar between, them, that is):

![aligned](https://user-images.githubusercontent.com/17967275/32750952-6bb1f32a-c881-11e7-8e6b-5ca4b67e5810.png)

I then tried rotating it via shift-e (and I hit the f key as you suggested, which then said radial symmetry around parent part):
![rotated](https://user-images.githubusercontent.com/17967275/32751002-8d58f7da-c881-11e7-818b-144a2e842349.png)

I see the problem here. I am trying to rotate so one of the 8 tanks is centered on the windows, which needs 4.5 clicks, which isn't a thing.

The sides go flat along 2 tanks, so rotating the normal 9 clicks puts you on the next face---with a gap where the legs would go.

The trick is that if you wish to use a bare tank, would the legs mount to the TANK, or mount to the gap?

With the awful stock legs, I want to mount to the tank, vs the gap. Better legs likely solve the problem.

**Comment by taterkerman** [comment]
More to the point, where I want to rotate the part is in fact a collider vertex, so it's gonna be one side or the other.

I'll just find some legs that look more at home on the gap.

**Comment by shadowmage45** [comment]
Erm.. maybe 'F' isn't the hotkey for local/global mode then.... (might be 'R'?  I hit it instinctively when in the editor, so I've forgotten which key it is exactly).  I'll get back to you with that information...

I can see the problem that you are running into when trying to mount the legs 'on a tank', as yes, that is where the corners of the colliders are located.

I'll try and come up with some screenshots/examples of how I do things (placing on the gap), and how they can be placed 'on the corner' (pretty sure it can be done).

**Comment by Aelfhe1m** [comment]
tater you could try the Editor Extensions mod for finer control over rotation and symmetry snap

**Comment by taterkerman** [comment]
"r" is the mirror one for aircraft stuff.

I think the problem will be that the tanks are where the vertices are of the collider, so they will not point the right way.

It;s not a big deal, you'd probably not put gear on a tank, anyway, it defeats the purpose of a lightweight tank, right?

We just need better gear that can attach to the framework.
![nasa_altair_lander_sm](https://user-images.githubusercontent.com/17967275/32755425-738d7f04-c892-11e7-901f-2825ffb689cc.jpg)

**Comment by shadowmage45** [comment]
Yeah, I do want to make up some 'lander' oriented landing legs.  Chances are they'll be specifically built to fit onto the new LV tanks.  Will probably be a stand-alone (single-leg) and 'quad' version, both will feature scaling, and should be constructed in such a way as to mostly just 'slot into' the tanks.

**Comment by taterkerman** [comment]
Yeah, not a huge deal. The Altair legs from that other mod are the right idea... I'll see if they work and drop a pic.

**Comment by taterkerman** [comment]
![legs](https://user-images.githubusercontent.com/17967275/32761076-aa6f538c-c8ae-11e7-9f97-216699fe1772.png)

**Comment by Jimbodiah** [comment]
Press the 2 key, offset them into the gap between the tanks. If you rotate them 22.5° then you will have issues placing a ladder on the tank sticking out (= looks fugly).

F is the right key. Also if you keep the L-Shift key presses while rotatin, you make smaller steps, but still can't center on the tanks exactly; tried it with 8x draco engines on the bottom ;)

**Comment by taterkerman** [comment]
Yeah, I realized my issue and posted above. Normally you'd rotate 5 clicks with left shift pressed. 1/2 of that is needed, but you cannot click 2.5 times ;)

**Comment by Jimbodiah** [comment]
I can actually, but KSP rounds up ;)

**Comment by shadowmage45** [comment]
I think the real 'solution' to this problem will come with the custom landing LV landing legs.  Which perhaps I need to start working on sooner rather than later.

@taterkerman you are right though -- it is nearly impossible to get the legs to lineup properly 'on a tank' -- even when using the all of the different offset modes, the stock tools simply refuse to snap at that 'half-an-increment' angle.

**Comment by Jimbodiah** [comment]
This effect is also evident when you try to line up something on a round fuel tank in 8-fold symmetry where you already have oher parts in 4-way symmetry and want to place them in the center of those.
